Include/internal/pycore_abstract.h

```diff
@@ -8,6 +8,16 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
+/* Return 1 if the specified index is outside the range[0, limit) */
+#define valid_index(index, limit) ((size_t)index < (size_t)limit)
```

Comment:
Please convert it to a static inline function and rename it to _Py_is_valid_index().

Reply:
Done! I also updated the description of the PR in the first comment.